components/net/image_cache_thread.rs

```diff
@@ -383,22 +345,21 @@ impl ImageCache {
             ImageCacheCommand::Exit(sender) => {
                 return Some(sender);
             }
-            ImageCacheCommand::RequestImage(url, result_chan, responder) => {
-                self.request_image(url, result_chan, responder, false);
-            }
-            ImageCacheCommand::RequestImageAndMetadata(url, result_chan, responder) => {
-                self.request_image(url, result_chan, responder, true);
-            }
-            ImageCacheCommand::GetImageIfAvailable(url, use_placeholder, consumer) => {
-                let result = self.get_image_if_available(url, use_placeholder);
-                let _ = consumer.send(result);
+            ImageCacheCommand::AddListener(id, responder) => {
+                self.add_listener(id, responder);
             }
-            ImageCacheCommand::GetImageOrMetadataIfAvailable(url, use_placeholder, consumer) => {
-                let result = self.get_image_or_meta_if_available(url, use_placeholder);
+            ImageCacheCommand::GetImageOrMetadataIfAvailable(url,
+                                                             use_placeholder,
+                                                             can_request,
+                                                             consumer) => {
+                let result = self.get_image_or_meta_if_available(url, use_placeholder, can_request);
                 let _ = consumer.send(result);
```

Comment:
Comment about when this can fail? (I mentioned this in several places, perhaps we could just document in one place when this would be expected to fail?).